File: src/main/scala/learning_flink/Review.scala
package main.scala.learning_flink

import org.apache.flink.api.scala.ExecutionEnvironment
import org.apache.flink.api.scala._

object Review {
  def main(args: Array[String]): Unit = {
    val env: ExecutionEnvironment = ExecutionEnvironment.getExecutionEnvironment
    val data: DataSet[String] = env.fromElements("Gift is better", "Money is best", "Money", "money")
    data.flatMap(_.toLowerCase.split(" "))
      .map((_, 1))
      .groupBy(0)
      .sum(1)
      .print()
  }

}

File: src/main/java/learning_flink_java/Review.java
package learning_flink_java;

import org.apache.flink.api.common.functions.FlatMapFunction;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.DataSource;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.util.Collector;

public class Review {
    public static void main(String[] args) throws Exception {
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
        DataSource<String> data = env.fromElements("Life is short I use Python Python", "Python is short");
        data.flatMap(new flatMapCls()).groupBy(0).sum(1).print();

    }

    public static class flatMapCls implements FlatMapFunction<String, Tuple2<String, Integer>> {
        @Override
        public void flatMap(String s, Collector<Tuple2<String, Integer>> collector) {
            String[] words = s.toLowerCase().split(" ");
            for (String word : words) {
                collector.collect(new Tuple2<>(word, 1));
            }
        }
    }
}
